### PR TITLE
Remove support for EOL versions of CDAP

### PR DESCRIPTION
**Pull request by wolf31o2**
CDAP 2.8 went EOL this month. Removing support for it and all earlier versions of CDAP from the cookbook.

- Removed 2.8.0 and 2.8.2 checksums
- Remove version check on `cdap-env.sh` template
- Remove version check on `logback-container.xml` copy
- Always include `ui` recipe from `fullstack` recipe
- Remove `cdap-gateway` service template/service resources
- Remove version check on `cdap-hbase-compat-0.98` package
- Remove `web_app` and `web_app_init` recipes
- Update specs

### CODE REVIEW

File: recipes/gateway.rb [hunk 1]
```diff
@@ -26,26 +26,19 @@
...
 include_recipe 'cdap::ssl_keystore_certificates'
 
-svcs = ['cdap-router']
-unless node['cdap']['version'].to_f >= 2.6
-  unless node['cdap']['version'].split('.')[2].to_i >= 9000
-    svcs += ['cdap-gateway']
-  end
-end
+svc = 'cdap-router'
+attrib = svc.gsub('cdap-', '').tr('-', '_')
```
Comment (dereklwood) [inline recipes/gateway.rb]: i don't think the ``.tr('-', '_')`` is needed anymore either.

**Comment by dereklwood** [comment]
minor suggestion, otherwise 👍 lgtm

File: recipes/gateway.rb (hunk 1 above)
Comment (dereklwood) [inline recipes/gateway.rb]: or just dont use vars for these strings for consistency with the other recipes.

**Comment by wolf31o2** [comment]
Additional changes:
- Removed `gateway` and `web_app` service attributes
- Minor `cdap_log_dir` cleanup in `default` recipe
- Get rid of variable use in gateway/router code, per PR feedback
- Deprecate `gateway` recipe in favor of `router`
- Guard `cdap-env.sh` creation since `node['cdap']['cdap_env']` may be null
- Use `cp -f` to avoid aliases

File: recipes/fullstack.rb [hunk 2]
```diff
@@ -19,8 +19,6 @@
 
 include_recipe 'cdap::default'
 
-ui_recipe = node['cdap']['version'].to_i < 3 ? 'web_app' : 'ui'
-
-%W(cli gateway kafka master security #{ui_recipe}).each do |recipe|
+%w(cli gateway kafka master security ui).each do |recipe|
```
Comment (dereklwood) [inline recipes/fullstack.rb]: router

**Comment by dereklwood** [comment]
small fix needed otherwise 👍 lgtm

**Pull request merged**